<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>Shaderkit::MainWindow</class>
 <widget class="QMainWindow" name="Shaderkit::MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>955</width>
    <height>732</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Shaderkit</string>
  </property>
  <property name="windowIcon">
   <iconset resource="../../gfx/gfx.qrc">
    <normaloff>:/icons/logo.png</normaloff>:/icons/logo.png</iconset>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QVBoxLayout" name="verticalLayout_3">
    <property name="spacing">
     <number>0</number>
    </property>
    <property name="margin">
     <number>0</number>
    </property>
    <item>
     <widget class="QTabWidget" name="editor_tabs">
      <property name="tabsClosable">
       <bool>true</bool>
      </property>
     </widget>
    </item>
    <item>
     <widget class="QWidget" name="editor_bar" native="true">
      <layout class="QVBoxLayout" name="verticalLayout_2">
       <property name="spacing">
        <number>0</number>
       </property>
       <property name="margin">
        <number>0</number>
       </property>
       <item>
        <widget class="QToolBar" name="editor_menu">
         <property name="iconSize">
          <size>
           <width>16</width>
           <height>16</height>
          </size>
         </property>
         <property name="floatable">
          <bool>false</bool>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>955</width>
     <height>21</height>
    </rect>
   </property>
   <widget class="QMenu" name="menu_file">
    <property name="title">
     <string>&amp;File</string>
    </property>
    <addaction name="action_newscene"/>
    <addaction name="action_new"/>
    <addaction name="action_open"/>
    <addaction name="separator"/>
    <addaction name="action_autosave_scene"/>
    <addaction name="action_savescene_as"/>
    <addaction name="action_savescene"/>
    <addaction name="action_savematerial"/>
    <addaction name="action_savefile"/>
    <addaction name="action_saveas"/>
    <addaction name="separator"/>
    <addaction name="action_import"/>
    <addaction name="action_export"/>
    <addaction name="separator"/>
    <addaction name="action_quit"/>
   </widget>
   <widget class="QMenu" name="menu_Help">
    <property name="title">
     <string>&amp;Help</string>
    </property>
    <addaction name="action_about"/>
   </widget>
   <widget class="QMenu" name="menuShader_Library">
    <property name="title">
     <string>Shader Library</string>
    </property>
    <addaction name="action_libopen"/>
    <addaction name="action_libexample"/>
   </widget>
   <widget class="QMenu" name="menuTools">
    <property name="title">
     <string>&amp;Tools</string>
    </property>
    <addaction name="action_sandbox_compiler"/>
    <addaction name="action_autosave_layout"/>
    <addaction name="separator"/>
    <addaction name="action_reload"/>
   </widget>
   <widget class="QMenu" name="menu_Window">
    <property name="title">
     <string>&amp;Window</string>
    </property>
    <addaction name="action_toggle_viewport"/>
    <addaction name="action_toggle_materials"/>
    <addaction name="action_toggle_render_passes"/>
    <addaction name="action_toggle_error_log"/>
    <addaction name="separator"/>
    <addaction name="action_reset_layout"/>
   </widget>
   <widget class="QMenu" name="menu_edit">
    <property name="title">
     <string>&amp;Edit</string>
    </property>
    <addaction name="action_textures"/>
   </widget>
   <addaction name="menu_file"/>
   <addaction name="menu_edit"/>
   <addaction name="menuShader_Library"/>
   <addaction name="menuTools"/>
   <addaction name="menu_Window"/>
   <addaction name="menu_Help"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <widget class="QDockWidget" name="error_dock">
   <property name="minimumSize">
    <size>
     <width>79</width>
     <height>26</height>
    </size>
   </property>
   <property name="windowTitle">
    <string>Compiler log</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>8</number>
   </attribute>
   <widget class="QFrame" name="frame">
    <property name="minimumSize">
     <size>
      <width>1</width>
      <height>1</height>
     </size>
    </property>
    <property name="frameShape">
     <enum>QFrame::StyledPanel</enum>
    </property>
    <property name="frameShadow">
     <enum>QFrame::Sunken</enum>
    </property>
    <layout class="QHBoxLayout" name="horizontalLayout">
     <property name="margin">
      <number>0</number>
     </property>
     <item>
      <widget class="QTableWidget" name="error_list">
       <property name="frameShape">
        <enum>QFrame::NoFrame</enum>
       </property>
       <property name="editTriggers">
        <set>QAbstractItemView::NoEditTriggers</set>
       </property>
       <property name="showDropIndicator" stdset="0">
        <bool>false</bool>
       </property>
       <property name="selectionMode">
        <enum>QAbstractItemView::SingleSelection</enum>
       </property>
       <property name="selectionBehavior">
        <enum>QAbstractItemView::SelectRows</enum>
       </property>
       <property name="verticalScrollMode">
        <enum>QAbstractItemView::ScrollPerPixel</enum>
       </property>
       <property name="horizontalScrollMode">
        <enum>QAbstractItemView::ScrollPerPixel</enum>
       </property>
       <property name="showGrid">
        <bool>true</bool>
       </property>
       <property name="gridStyle">
        <enum>Qt::DotLine</enum>
       </property>
       <property name="rowCount">
        <number>0</number>
       </property>
       <property name="columnCount">
        <number>3</number>
       </property>
       <attribute name="horizontalHeaderVisible">
        <bool>false</bool>
       </attribute>
       <attribute name="horizontalHeaderStretchLastSection">
        <bool>true</bool>
       </attribute>
       <attribute name="verticalHeaderVisible">
        <bool>false</bool>
       </attribute>
       <column/>
       <column/>
       <column/>
      </widget>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QDockWidget" name="materials_dock">
   <property name="minimumSize">
    <size>
     <width>79</width>
     <height>26</height>
    </size>
   </property>
   <property name="windowTitle">
    <string>Materials</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>4</number>
   </attribute>
   <widget class="QFrame" name="widget_2">
    <property name="minimumSize">
     <size>
      <width>1</width>
      <height>1</height>
     </size>
    </property>
    <property name="frameShape">
     <enum>QFrame::StyledPanel</enum>
    </property>
    <property name="frameShadow">
     <enum>QFrame::Sunken</enum>
    </property>
    <layout class="QVBoxLayout" name="verticalLayout_4">
     <property name="spacing">
      <number>0</number>
     </property>
     <property name="margin">
      <number>0</number>
     </property>
     <item>
      <widget class="MaterialProperties" name="material_properties"/>
     </item>
     <item>
      <widget class="QToolBar" name="m_prop_menu">
       <property name="iconSize">
        <size>
         <width>16</width>
         <height>16</height>
        </size>
       </property>
       <property name="floatable">
        <bool>false</bool>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QDockWidget" name="render_passes_dock">
   <property name="minimumSize">
    <size>
     <width>79</width>
     <height>26</height>
    </size>
   </property>
   <property name="features">
    <set>QDockWidget::AllDockWidgetFeatures</set>
   </property>
   <property name="windowTitle">
    <string>Render passes</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>4</number>
   </attribute>
   <widget class="QFrame" name="widget">
    <property name="minimumSize">
     <size>
      <width>1</width>
      <height>1</height>
     </size>
    </property>
    <property name="frameShape">
     <enum>QFrame::StyledPanel</enum>
    </property>
    <property name="frameShadow">
     <enum>QFrame::Sunken</enum>
    </property>
    <layout class="QVBoxLayout" name="verticalLayout">
     <property name="spacing">
      <number>0</number>
     </property>
     <property name="margin">
      <number>0</number>
     </property>
     <item>
      <widget class="RenderPassProperties" name="renderpass_properties"/>
     </item>
     <item>
      <widget class="QToolBar" name="rp_prop_menu">
       <property name="iconSize">
        <size>
         <width>16</width>
         <height>16</height>
        </size>
       </property>
       <property name="floatable">
        <bool>false</bool>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
  </widget>
  <action name="action_savematerial">
   <property name="text">
    <string>&amp;Save Material</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+S</string>
   </property>
  </action>
  <action name="action_open">
   <property name="text">
    <string>&amp;Open Project...</string>
   </property>
   <property name="toolTip">
    <string>Open</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+O</string>
   </property>
  </action>
  <action name="action_newscene">
   <property name="text">
    <string>&amp;New Project</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+N</string>
   </property>
  </action>
  <action name="action_saveas">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="text">
    <string>Save File &amp;As...</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Shift+S</string>
   </property>
   <property name="visible">
    <bool>false</bool>
   </property>
  </action>
  <action name="action_quit">
   <property name="text">
    <string>&amp;Quit</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Q</string>
   </property>
  </action>
  <action name="action_about">
   <property name="text">
    <string>&amp;About</string>
   </property>
  </action>
  <action name="action_new">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="text">
    <string>New &amp;File</string>
   </property>
  </action>
  <action name="action_savescene">
   <property name="text">
    <string>Save Project</string>
   </property>
  </action>
  <action name="action_export">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="text">
    <string>Export...</string>
   </property>
   <property name="visible">
    <bool>false</bool>
   </property>
  </action>
  <action name="action_libexample">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="text">
    <string>Open Example</string>
   </property>
  </action>
  <action name="action_libopen">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="text">
    <string>Open Library</string>
   </property>
   <property name="visible">
    <bool>false</bool>
   </property>
  </action>
  <action name="action_autosave">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="checked">
    <bool>true</bool>
   </property>
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="text">
    <string>Autosave Project</string>
   </property>
  </action>
  <action name="action_toggle_viewport">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="checked">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>OpenGL viewport</string>
   </property>
  </action>
  <action name="action_toggle_materials">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="checked">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Materials</string>
   </property>
   <property name="iconText">
    <string>Materials</string>
   </property>
   <property name="toolTip">
    <string>Materials</string>
   </property>
  </action>
  <action name="action_toggle_error_log">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="checked">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Error log</string>
   </property>
  </action>
  <action name="action_autosave_layout">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="checked">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Autosave Layout</string>
   </property>
  </action>
  <action name="action_reset_layout">
   <property name="text">
    <string>Reset Layout</string>
   </property>
  </action>
  <action name="action_toggle_render_passes">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="checked">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Render passes</string>
   </property>
   <property name="toolTip">
    <string>Render passes</string>
   </property>
  </action>
  <action name="action_reload">
   <property name="text">
    <string>&amp;Reload Project File</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+R</string>
   </property>
  </action>
  <action name="action_sandbox_compiler">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="checked">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>&amp;Sandbox Compiler</string>
   </property>
   <property name="toolTip">
    <string>Use Sandbox Compiler</string>
   </property>
  </action>
  <action name="action_import">
   <property name="text">
    <string>&amp;Import...</string>
   </property>
  </action>
  <action name="action_textures">
   <property name="icon">
    <iconset resource="../../gfx/gfx.qrc">
     <normaloff>:/icons/texture_browser.png</normaloff>:/icons/texture_browser.png</iconset>
   </property>
   <property name="text">
    <string>&amp;Textures</string>
   </property>
  </action>
  <action name="action_savefile">
   <property name="text">
    <string>Save Active File</string>
   </property>
   <property name="visible">
    <bool>false</bool>
   </property>
  </action>
  <action name="action_autosave_scene">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="checked">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Autosave Project</string>
   </property>
  </action>
  <action name="action_savescene_as">
   <property name="text">
    <string>Save Project As...</string>
   </property>
  </action>
 </widget>
 <customwidgets>
  <customwidget>
   <class>MaterialProperties</class>
   <extends>QTableWidget</extends>
  </customwidget>
  <customwidget>
   <class>RenderPassProperties</class>
   <extends>QTableWidget</extends>
   <header>gui/render_pass_properties.hpp</header>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="../../gfx/gfx.qrc"/>
 </resources>
 <connections/>
</ui>
